### PR TITLE
Update info about the current NumPy teams and institutional partners

### PR DESCRIPTION
**Pull request by InessaPawson**
- Update info about the current NumPy teams and institutional partners.
- Rephrase for clarity some parts of the text.

### CODE REVIEW

File: content/en/about.md [hunk 1]
```diff
@@ -67,8 +68,7 @@ NumPy receives direct funding from the following sources:
 
 Institutional Partners are organizations that support the project by employing people that contribute to NumPy as part of their job. Current Institutional Partners include:
 
-- UC Berkeley (Stéfan van der Walt, Ross Barnowski)
```
Comment (stefanv) [inline content/en/about.md:70]: I was going to approve this, but then remembered that I am still hired by Berkeley :)

**Review by stefanv — APPROVED** [review]

File: content/en/about.md [hunk 2]
```diff
@@ -38,16 +36,19 @@ Emeritus:
 
 ## Teams
 
-The NumPy project is growing! &#127881; We have teams for:
+The NumPy project leadership is actively working on diversifying contribution pathways to the project.<br>
+NumPy currently has the following teams:
 
-- code
+- developer
```
[the remark this answers is not in the document]
Comment (InessaPawson) [reply content/en/about.md:42]: Good point, @melissawm! I'll make the change.

Comment (InessaPawson) [inline content/en/about.md:42]:
```suggestion
- development
```

Comment (stefanv) [reply content/en/about.md:42]: I saw that, but I thought the names corresponded to GitHub team names.

Comment (stefanv) [reply content/en/about.md:42]: I guess not! https://github.com/orgs/numpy/teams

**Pull request merged**